packages/service-library/src/servicelib/redis.py

```diff
@@ -223,17 +223,20 @@ class RedisClientsManager:
     databases: set[RedisDatabase]
     settings: RedisSettings
 
-    _client_sdks: dict[RedisDatabase, RedisClientSDK] = field(default_factory=dict)
+    _client_sdks: dict[RedisDatabase, RedisClientSDKHealthChecked] = field(
+        default_factory=dict
+    )
 
     async def setup(self) -> None:
         for db in self.databases:
-            self._client_sdks[db] = client_sdk = RedisClientSDK(
+            self._client_sdks[db] = RedisClientSDKHealthChecked(
                 redis_dsn=self.settings.build_redis_dsn(db)
             )
-            await client_sdk.setup()
+
+        await logged_gather(*(c.setup() for c in self._client_sdks.values()))
```

Comment:
you do not check the return values here? who closes the non-broken clients?
we usually retry these things on start, that means that if one client fails to connect (which might very well happen say if the network is being set up), then only a part of the clients will be created. 
Also, I think there is already a heatlhcheck mechanism used at several places, that runs in the background and collects different health.

Reply:
I must have gotten the wrong impression, thought this would raise if one of the stop connection fails.
If setup fails the app boot will stop. I will check this once more

Reply:
so then how do you close them opened clients?
say I have 2 databases, the 1st one fails and the 2nd one fails. who closes the 2nd one?

Reply:
Q: And here we have N background tasks pinging to the same url?

Reply:
So, I've done a bit of digging. If you call `shutdown()` on a client that has not yet called `setup()` it does not raise an error. Just expanded the tests to make sure this does not change.

By design you call shutdown on all clients. I guess that was my original intention here.

I have also changed the logged gather to a sequential since it does not play nice with Redis, especially when closing everything.